rust/src/stressers/sensor/mod.rs

```diff
@@ -0,0 +1,97 @@
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use anyhow::Result;
+use log::*;
+use tokio::time::sleep;
+
+use crate::cli::ServerConfig;
+use crate::database;
+use crate::model::owner::Owner;
+use crate::model::pet::Pet;
+use crate::model::sensors::sensor::Sensor;
+use crate::model::sensors::sensor_measure::Measure;
+use crate::repositories::Repositories;
+
+pub async fn sensor_stress(config: &ServerConfig, measure: &Duration, buffer_interval: &Duration) -> Result<()> {
+    info!("Welcome to the Pet collar simulator");
+
+    let sess = Arc::new(database::new_session_with_keyspace(&config).await?);
+    let repositories = Repositories::new(Arc::clone(&sess)).await;
+
+    let (owner, pet, sensors) = random_data();
+    save_data(&repositories, &owner, &pet, &sensors).await?;
+    run_sensor_data(&repositories, sensors, measure, buffer_interval).await?;
+
+    Ok(())
+}
+
+async fn run_sensor_data(
+    repositories: &Repositories,
+    sensors: Vec<Sensor>,
+    measure: &Duration,
+    buffer_interval: &Duration
+) -> Result<()> {
+
+    let mut last = Instant::now();
+    loop {
+        let mut measures = vec![];
+        while last.elapsed() < *buffer_interval {
+            sleep(measure.clone()).await;
```

Comment:
```suggestion
           sleep(*measure).await;
```
Duration implements Clone so you can deref here

Reply:
If `Duration` implements `Clone`, he can call `clone()`.
If `Duration` implements `Copy`, he can copy it without calling `clone()`, e.g. by dereferencing.